### DISCUSSION TITLE
How to troubleshooting outbound email problems?

### DISCUSSION THREAD

**Discussion by DANgerous25** [simple-login/app#1188] [Q&A]
Hi all, I’ve successfully set up a self-host setup. I have a few domains, but for simplicity let’s call call them domain1.xyz, domain2.xyz and domain3.abc. My self-host is set up on app.domain1.xyz. It is able to send outbound email to recipients of domain1.xyz and domain2.xyz, but not domain3.abc. I’m puzzled as to why domain3.abc doesn’t receive the email. 

Short question: How do I troubleshoot why email doesn’t send? I don’t get any feedback from the SL web app, mail appears to send (specifically the sign up activation email) but never arrives. 

I managed to find a way to look at some logs, and it appears the mail is getting bounced. I’m guessing it is something to do with dmarc/spf/dkim on domain3.abc. But I don’t interpret a lot of info from the log. Example:

`$docker logs sl-email`

`2022-07-26 04:56:49,550 - SL - INFO - "email_handler.py:1634" - _handle() - 58daa735-0d5b-4fbf-8c9e-c8ce57ec8a1d - ===>> New message, mail from <>, rctp tos ['transactional+5+@domain1.xyz'] 
set message_id dc8a6701-b32d-4978-ac58-44bbd8116a19
2022-07-26 04:56:49,556 - SL - DEBUG - "email_handler.py:1543" - handle() - dc8a6701-b32d-4978-ac58-44bbd8116a19 - ==>> Handle mail_from:<>, rcpt_tos:['transactional+5+@domain1.xyz'], header_from:MAILER-DAEMON@falconar.xyz (Mail Delivery System), header_to:transactional+5+@domain1.xyz, cc:None, reply-to:None, mail_options:['SIZE=24106', 'BODY=8BITMIME'], rcpt_options:[]
2022-07-26 04:56:49,575 - SL - DEBUG - "email_handler.py:1566" - handle() - dc8a6701-b32d-4978-ac58-44bbd8116a19 - Handle email sent to sender from <>
2022-07-26 04:56:49,575 - SL - DEBUG - "email_handler.py:1446" - handle_transactional_bounce() - dc8a6701-b32d-4978-ac58-44bbd8116a19 - handle transactional bounce sent to transactional+5+@domain1.xyz
2022-07-26 04:56:49,578 - SL - INFO - "email_handler.py:1454" - handle_transactional_bounce() - dc8a6701-b32d-4978-ac58-44bbd8116a19 - Create bounce for myrecipient@domain3.abc
2022-07-26 04:56:49,587 - SL - INFO - "email_handler.py:1644" - _handle() - dc8a6701-b32d-4978-ac58-44bbd8116a19 - takes 0.036536455154418945 seconds <<===
`

It looks like there is some sort of bounce email, but I don’t know how to look at it and inspect its headers. Is this possible? Is it possible to send a test mail and view the headers? I followed the guide to send from swaks, which shows a rejection but no much info as to why (error = `** 550 5.7.1 [NR] Message rejected`) - plus I learned that swaks doesn’t yet include the dkim key which might be the culprit. I also tried various online tools to test dmarc/spf/dkim and they say everything is ok with app.domain1.xyz.

Thanks all for your help, please let me know what additional info I can provide.

**Comment by nguyenkims** [comment]
@DANgerous25 I'd suggest to take a look at Postfix log, usually at `/var/log/mail.log` as it can contain the reason why the email was bounced.

**Reply by DANgerous25** [reply]
Thank you for that suggestion, it does seem to give some additional detail but I’ve no idea what it really means:

Jul 26 09:23:57 app postfix/smtpd[107957]: connect from unknown[10.0.0.3]
Jul 26 09:23:57 app postfix/smtpd[107957]: B346E46F53: client=unknown[10.0.0.3]
Jul 26 09:23:57 app postfix/cleanup[107963]: B346E46F53: message-id=<165882743767.10.15927548402211493405@0229db402188>
Jul 26 09:23:57 app postfix/qmgr[107925]: B346E46F53: from=<transactional+2+@mydomain1.xyz>, size=22079, nrcpt=1 (queue active)
Jul 26 09:23:57 app postfix/smtpd[107957]: disconnect from unknown[10.0.0.3] ehlo=1 mail=1 rcpt=1 data=1 commands=4
Jul 26 09:24:00 app postfix/smtp[107964]: B346E46F53: to=<myrecipient@domain3.abc>, relay=mx2.myhostingrelay.net[ip address]:25, delay=3.1, delays=0.03/0.02/0.93/2.1, dsn=5.7.1, status=bounced (host mx2.myhostingrelay.net[ip address] said: 550 5.7.1 [NR] Message rejected (in reply to end of DATA command))
Jul 26 09:24:00 app postfix/cleanup[107963]: D0A6046F55: message-id=<20220726092400.D0A6046F55@app.mydomain1.xyz>
Jul 26 09:24:00 app postfix/bounce[107966]: B346E46F53: sender non-delivery notification: D0A6046F55
Jul 26 09:24:00 app postfix/qmgr[107925]: D0A6046F55: from=<>, size=24101, nrcpt=1 (queue active)
Jul 26 09:24:00 app postfix/qmgr[107925]: B346E46F53: removed
Jul 26 09:24:00 app postfix/smtp[107964]: D0A6046F55: to=<transactional+2+@mydomain1.xyz>, relay=127.0.0.1[127.0.0.1]:20381, delay=0.07, delays=0.01/0.01/0/0.04, dsn=2.0.0, status=sent (250 bounce handled)
Jul 26 09:24:00 app postfix/qmgr[107925]: D0A6046F55: removed

Is there an issue because 10.0.0.3 is “unknown” and if so, how can I fix that? Or something else.

**Reply by nguyenkims** [reply]
The important bit is `550 5.7.1 [NR] Message rejected (in reply to end of DATA command))`. This error message depends on the mailbox provider, I suggest to check with the mailbox provider on how to solve this error.
